src/stan/io/empty_var_context.hpp

```diff
@@ -0,0 +1,113 @@
+#ifndef STAN_IO_EMPTY_VAR_CONTEXT_HPP
+#define STAN_IO_EMPTY_VAR_CONTEXT_HPP
+
+#include <stan/io/var_context.hpp>
+#include <string>
+#include <vector>
+
+namespace stan {
+  namespace io {
+
+    /**
+     * This is an implementation of a var_context that doesn't contain
+     * any variables.
+     */
+    class empty_var_context : public var_context {
+    public:
+      /**
+       * Destructor
+       *
+       */
+      ~empty_var_context() {}
```

Comment:
I'll make sure the base class, `var_context`, has a virtual destructor. As far as I can tell, the subclass's destructor doesn't have to be virtual. I don't think there will be a measurable hit if there's a performance difference: we only create this once in CmdStan for the lifetime of the program.